python/paddle/fluid/dygraph/varbase_patch_methods.py

```diff
@@ -559,11 +559,29 @@ def contain_tensor(item):
                         return True
             return False
 
-        if contain_tensor(item):
+        def is_list_index(item):
+            def _is_list(item):
+                if not isinstance(item, (list, tuple, int, np.ndarray)):
+                    return False
+                if isinstance(item, (tuple, list)):
+                    for s in item:
+                        if not _is_list(s):
+                            return False
+                return True
+
+            if isinstance(item, np.ndarray):
+                return True
+            if not isinstance(item, (tuple, list)):
```

Comment:
do we need to support `set`?

Reply:
revert the modification of index_select_op.